internal/pipeline/sign_test.go

```diff
@@ -51,19 +51,19 @@ func TestSignVerify(t *testing.T) {
 			name:                           "HMAC-SHA256",
 			generateSigner:                 func(alg jwa.SignatureAlgorithm) (jwk.Key, jwk.Set) { return newSymmetricKeyPair(t, "alpacas", alg) },
 			alg:                            jwa.HS256,
-			expectedDeterministicSignature: "eyJhbGciOiJIUzI1NiIsImtpZCI6IlRlc3RTaWduVmVyaWZ5In0..Xd7udcMRc3Gg236JdiV2vggGrqxAfgfLZdCLUpgAN34",
+			expectedDeterministicSignature: "eyJhbGciOiJIUzI1NiIsImtpZCI6IlRlc3RTaWduVmVyaWZ5In0..codTLtvPLnSEetO6RIaYBg0bspzJp4lNDQ5XzU606wQ",
 		},
 		{
 			name:                           "HMAC-SHA384",
 			generateSigner:                 func(alg jwa.SignatureAlgorithm) (jwk.Key, jwk.Set) { return newSymmetricKeyPair(t, "alpacas", alg) },
 			alg:                            jwa.HS384,
-			expectedDeterministicSignature: "eyJhbGciOiJIUzM4NCIsImtpZCI6IlRlc3RTaWduVmVyaWZ5In0..g-_B2RO6o_oZjPoM2UyCHDANbPeeqLBUexLRl_MoW7BdpLC7r6mLc0wgRIzJy6ih",
+			expectedDeterministicSignature: "eyJhbGciOiJIUzM4NCIsImtpZCI6IlRlc3RTaWduVmVyaWZ5In0..HhRzgsJXO2eyaE-ldmw3FQh_0Zl-Ral080EV4zqnch3b7N-2Exn3gTvyjOM6clPV",
 		},
 		{
 			name:                           "HMAC-SHA512",
 			generateSigner:                 func(alg jwa.SignatureAlgorithm) (jwk.Key, jwk.Set) { return newSymmetricKeyPair(t, "alpacas", alg) },
 			alg:                            jwa.HS512,
-			expectedDeterministicSignature: "eyJhbGciOiJIUzUxMiIsImtpZCI6IlRlc3RTaWduVmVyaWZ5In0..iW8eaMBrcK7Ehj41DRzgQp3haYBf70JgA_n0C4d_acRZCdVUm-GJv9pdxQ5O0pYd7gJC_wMmaNMkuj4TXqlPvg",
+			expectedDeterministicSignature: "eyJhbGciOiJIUzUxMiIsImtpZCI6IlRlc3RTaWduVmVyaWZ5In0..Ku9vjRiYsc4Wy4Rgfw4fzB70_qFM21bPrrVfqSmgGBWqoR1ZzVsjoluUQ5eBEEEhYZvI9QlbwAQtqXOwrS2w5A",
```

Comment:
It's definitely a change detector. I haven't minded so much, because every time we have to fix these, we've broken compatibility between agent versions. That's going to be a problem for people with mixed versions, but I won't feel bad telling those people not to do that. But it could become a problem for "static" signatures lying around (e.g. in the Pipeline YAML on the backend side).

Reply:
> But it could become a problem for "static" signatures lying around (e.g. in the Pipeline YAML on the backend side).

i hadn't considered that at all! that's a really good point.

one would hope that once we nail down the implementation of our signatures (which feels like it's getting close), the signature format(? if that's the right word?) won't change nearly as often as it currently is

Reply:
We are in the semver `0.x.x` stage for signature versions. Once we hit `1.x.x` we should not break compatibility.